fs-storage/src/folder_storage.rs

```diff
@@ -595,4 +589,260 @@ mod tests {
         assert_eq!(storage1.as_ref().get("key2"), Some(&6));
         assert_eq!(storage1.as_ref().get("key3"), Some(&9));
     }
+
+    use data_error::Result;
+    use quickcheck::{Arbitrary, Gen};
+    use std::collections::{BTreeMap, HashSet};
+    use std::time::SystemTime;
+
+    // Assuming FolderStorage, BaseStorage, SyncStatus, and other necessary types are in scope
+
+    #[derive(Clone, Debug)]
+    enum StorageOperation {
```

Comment:
Is it possible to re-use the enum from `FileStorage` or operations differ indeed?

Reply:
Unfortunately, these are different operations.